### PR TITLE
docs: add issue 749 BC warm-start launch packet

### LINKED ISSUE #749: research: train a BC-preinitialized PPO challenger under the v10 fine-tune contract
## Goal / Problem

**Research question**
Can behavior-cloning pretraining from the promoted PPO v3 expert policy produce a PPO fine-tuning run that is either more sample efficient or benchmark-stronger than the current carry-forward candidate `27dbe5xu`?

**Hypothesis**
A BC-initialized policy will preserve stronger route priors early in training and may reduce obstacle-collision and `max_steps` failures that remain in pure PPO continuation runs.

**Scientific motivation**
Issue 739 showed that reward and observation tweaks were not the right path. The imitation pipeline is the main materially different PPO-improvement route already supported by this repository.

## Scope

- Independent variables: initialization strategy (`BC pretrain -> PPO fine-tune`) under the existing v10 observation/reward/predictive-foresight contract.
- Dependent variables: timesteps to strong eval checkpoints, full policy-analysis success/collision/`max_steps`/SNQI, and final benchmark comparability.
- Control variables: benchmark evaluation surface, promoted v3 expert source, and the downstream PPO fine-tune contract.
- In scope: collect or validate the expert-trajectory dataset, run BC pretraining, fine-tune with PPO, compare against `27dbe5xu` and `b60iopxt`, and record the result conservatively.
- Out of scope: new reward design, new architectures beyond the existing imitation pipeline, or broad hyperparameter sweeps.

## Added Value Estimation

- Research value: High. This is the strongest non-reward PPO direction already supported by repo tooling.
- Benchmark value: High. A positive result could create a materially different PPO candidate family.
- Publication value: Medium-High. It would strengthen the narrative around sample efficiency and warm starts.
- Why now: The direct retrain path regressed, so the next serious PPO attempt should explore a supported path with a different inductive bias.

## Effort Estimation

- Rough estimate (hours): 10
- Best estimate (hours): 14
- Unknowns: dataset quality, imitation dependency setup, BC stability, and the extra runtime from a multi-stage pipeline.

## Complexity Estimation

- Experimental complexity: High.
- Dependencies: `docs/imitation_learning_pipeline.md`, `configs/training/ppo_imitation/`, expert trajectory collection, BC pretraining scripts, PPO fine-tune scripts, W&B tracking, and benchmark evaluation scripts.
- Open questions: whether the promoted v3 expert trajectories transfer cleanly to the v10 predictive-foresight fine-tune contract.

## Risk Assessment

- Statistical risk: Medium. A single BC seed may not represent the family well.
- Reproducibility risk: Medium-High. This is a multi-stage artifact chain that depends on dataset lineage.
- Runtime risk: High. Dataset collection, BC, PPO fine-tuning, and full policy analysis all add runtime.
- Mitigation: keep the pipeline config-first, validate datasets/manifests, and compare only against the maintained PPO candidate surfaces.

## Affected Files

- `configs/training/ppo_imitation/` - Add or update BC-pretrain and PPO fine-tune configs for the challenger run.
- `output/benchmarks/expert_trajectories/` - Store dataset artifacts and manifests.
- `output/imitation_reports/` - Store BC/PPO run manifests and comparisons.
- `docs/context/` - Record a conservative candidate comparison note.

## Definition of Done

- [ ] The expert trajectory dataset is versioned and validated.
- [ ] A BC-pretrained checkpoint is produced with reproducible config metadata.
- [ ] PPO fine-tuning runs under the maintained v10 contract.
- [ ] The selected checkpoint is compared against `27dbe5xu` and `b60iopxt` on the canonical evaluation path.
- [ ] The result is documented with explicit caveats on sample efficiency and benchmark strength.

## Success Metrics

- The end-to-end pipeline is reproducible from committed configs.
- The challenger reaches equal or better benchmark-facing quality than `27dbe5xu`, or the negative result is documented clearly.
- Sample-efficiency evidence is recorded rather than inferred.

## Validation / Testing

- [ ] Run expert trajectory collection or validate an existing approved dataset.
- [ ] Run `uv run --group imitation python scripts/training/pretrain_from_expert.py --config <bc-config>`.
- [ ] Run `uv run python scripts/training/train_ppo_with_pretrained_policy.py --config <ppo-finetune-config>`.
- [ ] Run the canonical policy-analysis and candidate-comparison evaluation path.

## Estimate Discussion

- Why these values were proposed: the pipeline already exists, but unlike the `num_envs` path it has not yet been established as benchmark-grade in this checkout, so the scope includes more setup and more risk.
- Uncertainty / confidence: Medium confidence in the workflow, lower confidence in the outcome.
- What evidence would move the estimate: pre-existing approved datasets, smooth BC convergence, or early evidence that transfer to the v10 contract is poor.

## Project Metadata

- Priority: Medium
- Effort (h): 14
- Reviewed: 2026-05-09

### PR DESCRIPTION
**Pull request by ll7**
## Summary
Adds a config-first launch packet for the #749 BC-preinitialized PPO challenger and records the current evidence gap: no durable #749 run/checkpoint chain exists yet, so this PR does not claim an experiment result.

## Linked Issues
- Relates to #749
- Follow-up execution issue: #1108

## What Changed
- Added issue-specific BC and PPO fine-tune configs:
  - `configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml`
  - `configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml`
- Added `docs/context/issue_749_bc_preinit_ppo_launch_packet.md` with canonical commands, comparator IDs, artifact boundaries, and dry-run validation scope.
- Linked the launch packet from `docs/README.md`, `docs/context/README.md`, and `configs/training/ppo_imitation/README.md`.
- Fixed stale imitation spec text that passed unsupported `--dataset` to `pretrain_from_expert.py`.
- Added a config-loading test that proves the #749 BC policy output ID feeds the PPO fine-tune `pretrained_policy_id`.

## Why It Matters
- Added value: #749 now has a reproducible branchable launch path instead of only a broad experiment request.
- Expected impact: the expensive BC/PPO run can be launched and reviewed without inventing config names or artifact IDs later.
- Why this is worth merging now: tracked docs already say no #749 W&B run was found; this PR turns that gap into an executable plan and opens #1108 for the actual long-running work.

## Validation / Proof
- Commands run:
  - `uv run python scripts/training/collect_expert_trajectories.py --dataset-id issue_749_b60iopxt_v10_eval_trajectories --policy-id ppo_expert_br06_v3_15m_all_maps_randomized_20260304T075200 --episodes 1 --scenario-config configs/scenarios/sets/ppo_full_maintained_eval_v1.yaml --seeds 111 --dry-run`
  - `uv run --group imitation python scripts/training/pretrain_from_expert.py --config configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml --dry-run`
  - `uv run python scripts/training/train_ppo_with_pretrained_policy.py --config configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml --dry-run`
  - `uv run pytest tests/integration/test_ppo_pretraining_pipeline.py tests/training/test_pretrain_from_expert.py -q` -> `10 passed in 6.07s`
  - `rtk rg -n -- "pretrain_from_expert.py --dataset" specs docs scripts tests || true`
  - `rtk git diff --check`
  - `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main rtk scripts/dev/pr_ready_check.sh` -> `3272 passed, 18 skipped, 3 warnings in 255.28s`
  - `BASE_REF=origin/main rtk uv run python scripts/dev/pr_ready_freshness.py write --base-ref origin/main`
- Evidence that the change works here: the dry-run path generated the expected ignored dataset, BC checkpoint, PPO fine-tune checkpoint, and manifests under `output/`; the new test validates config linkage without relying on generated artifacts.
- Benchmarks or smoke tests, if applicable: no benchmark result is claimed; dry-run artifacts are launch-path proof only.

## Risks / Rollout
- Compatibility risks: low; this adds configs/docs/tests and fixes stale spec command text.
- Failure modes: users could mistake dry-run outputs for #749 evidence, so the context note explicitly forbids that.
- Rollback or fallback plan: revert this docs/config commit if a different #749 experiment contract is preferred.

## Docs / Provenance
- Updated docs: `docs/context/issue_749_bc_preinit_ppo_launch_packet.md`, `docs/README.md`, `docs/context/README.md`, `configs/training/ppo_imitation/README.md`, `specs/001-ppo-imitation-pretrain/tasks.md`.
- Relevant design or provenance notes: `docs/context/issue_708_main_based_ppo_retrain_campaign.md`, `docs/imitation_learning_pipeline.md`, `model/registry.yaml`.
- Any assumptions that need to be preserved: `b60iopxt` is the historical promoted v3 expert source; `27dbe5xu` remains the v10 carry-forward comparator; full experiment execution is deferred to #1108.

## Follow-Up Issues
- Deferred work: real dataset collection, BC pretraining, PPO fine-tuning, durable artifact promotion, and policy-analysis comparison.
- Issues opened for follow-up: #1108.

## Reviewer Notes
- Anything a reviewer should verify closely: whether the #749 run IDs and 10M fine-tune budget match the intended campaign naming and scale.
- Any known limitations: this PR does not close #749 because it does not produce the real BC/PPO artifacts or comparison result.

### CODE REVIEW

**Comment by ll7** [comment]
Addressed the remaining review point on `tests/integration/test_ppo_pretraining_pipeline.py`.

What changed:
- Replaced cwd-relative config paths with `Path(__file__).resolve().parents[2]` so the new `#749` warm-start linkage test resolves repo files from the test location instead of the runner's working directory.

Implementation verification against issue #749 and PR #1109 at `163485466b80f142f77a92e88fec315832429b7d`:

| Claim | Evidence surface | Validation command/artifact | Result | Residual risk |
| --- | --- | --- | --- | --- |
| Issue-specific BC and PPO warm-start configs exist for the `#749` launch path | `configs/training/ppo_imitation/bc_pretrain_issue_749_v10_warm_start.yaml`, `configs/training/ppo_imitation/ppo_finetune_issue_749_v10_warm_start.yaml` | Branch diff vs `origin/main`; targeted linkage test | Observed | This still does not produce the real dataset/checkpoint chain from issue #749 |
| The BC output ID feeds the PPO fine-tune config through a robust test path | `tests/integration/test_ppo_pretraining_pipeline.py` | `uv run pytest tests/integration/test_ppo_pretraining_pipeline.py -q` | Observed pass after this fix | The test proves config linkage, not full training execution |
| The branch remains a conservative launch packet rather than an overclaimed experiment result | `docs/context/issue_749_bc_preinit_ppo_launch_packet.md`, PR body, issue #749 body | Manual issue/PR-to-branch comparison | Observed | Actual dataset collection, BC pretraining, PPO fine-tuning, and policy analysis remain deferred to #1108 |
| The branch remains repo-ready after the review fix | Full branch against `origin/main` | `PYTEST_NUM_WORKERS=8 BASE_REF=origin/main scripts/dev/pr_ready_check.sh` | Observed pass: `3273 passed, 17 skipped, 3 warnings` | Same standing warnings as before; no new failures introduced |

Artifact persistence decision:
- Ignored `output/*` content still includes local dry-run and benchmark byproducts. This review fix did not create any new durable artifact dependency, and the branch still correctly treats dry-run outputs as non-evidence for issue #749.
